x/qgb/orchestrator/orchestrator.go

```diff
@@ -4,95 +4,436 @@ import (
 	"context"
 	"crypto/ecdsa"
 	"fmt"
+	paytypes "github.com/celestiaorg/celestia-app/x/payment/types"
+	"github.com/celestiaorg/celestia-app/x/qgb/types"
 	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdktypestx "github.com/cosmos/cosmos-sdk/types/tx"
 	stakingtypes "github.com/cosmos/cosmos-sdk/x/staking/types"
-	"github.com/tendermint/tendermint/libs/log"
-	"math/big"
-
-	"github.com/celestiaorg/celestia-app/x/qgb/types"
 	ethcmn "github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/pkg/errors"
+	tmlog "github.com/tendermint/tendermint/libs/log"
+	corerpctypes "github.com/tendermint/tendermint/rpc/core/types"
+	coretypes "github.com/tendermint/tendermint/types"
+	"google.golang.org/grpc"
+	"math/big"
+	"strconv"
+	"sync"
+	"time"
 )
 
-type orchestrator struct {
-	logger log.Logger
-	// TODO this will change once we have the worker pool pattern
+type Orchestrator struct {
+	logger tmlog.Logger // maybe use a more general interface
```

Comment:
@evan-forbes should we use a more generic logger interface? tendermint is quite opinionated, no?

Reply:
the reasoning behind using the tendermint logger was just to stay consistent with what is used in the app. While I certainly agree that a different logger would probably fit our case better, using two different loggers in the same repo sounds like not a good idea. If we end up moving the orchestrator/relayer out of this repo, then I'd feel a lot better about switching it.

Reply:
Understandable :+1: